File: Java/0001-0050/9. Palindrome Number.java
// https://leetcode.com/problems/palindrome-number/description/

class Solution {
    public boolean isPalindrome(int x) {
        int reversedInt = 0,copy = x;

        //reversing the given int to a copy variable
        while(copy>0){
            //(%10) to get last digit of a number
            reversedInt = (reversedInt*10) + copy%10;
            //after getting last digit remove the last digit by (/10)
            copy/=10;
        }

        return (x == reversedInt) ? true : false; //if reversed and given number is same its palindrome
    }
}

File: Python/0001-0050/9. Palindrome Number.py
# https://leetcode.com/problems/palindrome-number/description/

class Solution:
    def isPalindrome(self, x: int) -> bool:
        reversedInt, copy = 0, x

        # reversing the given int to a copy variable
        while copy>0:
            # (%10) to get last digit of a number
            reversedInt = (reversedInt*10) + copy%10
            # after getting last digit remove the last digit by (/10)
            copy//=10
        
        # //if reversed and given number is same its palindrome
        return True if (x == reversedInt) else False
